apis/stargate/v1alpha1/stargate_types.go

```diff
@@ -151,16 +152,20 @@ type StargateDatacenterTemplate struct {
 	Racks []StargateRackTemplate `json:"racks,omitempty"`
 }
 
-// Coalesce compares this StargateDatacenterTemplate with the given StargateClusterTemplate and returns the first
-// non-nil StargateDatacenterTemplate it finds.
-// TODO revisit the merging strategy and/or find a better name for this method
-func (in *StargateDatacenterTemplate) Coalesce(clusterTemplate *StargateClusterTemplate) *StargateDatacenterTemplate {
+// MergeWith merges the given StargateClusterTemplate with this StargateDatacenterTemplate and
+// returns the result.
+func (in *StargateDatacenterTemplate) MergeWith(clusterTemplate *StargateClusterTemplate) *StargateDatacenterTemplate {
```

Comment:
I had to standardize method names and the general contract. I went for `MergeWith` and the contract is that the receiver is the DC level object, and the passed object is the cluster level one.